Marlin/clean_nozzle.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef __CLEAN_NOZZLE_H__
+#define __CLEAN_NOZZLE_H__
+
+#include "point_t.h"
+
+extern void do_blocking_move_to(float x, float y, float z);
```

Comment:
The first iteration I did was static but not knowing the future reverted back to non-static as the binary size was the same (we may have four nozzles, etc) but Ido agree with your principle for static and we should make clear that will be the norm.

I believe there is only one road to OOP, we must break with the past. I may be over killing with dynamic classes but the reason for that is that we dont know know the future or what we will require. The problem is every new class is bloated because old code does not re-use them.

GCC also inlines methods, there is no reason `movement` to be restrained in `Marlin_main.cpp`. IMO we must change the mindset that everything must be optimized, we must optimize code that runs most of the time for speed and the other part for size (80-20 rule).